modules/stochastic_tools/src/vectorpostprocessors/SamplerData.C

```diff
@@ -0,0 +1,56 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+// Stocastic Tools Includes
+#include "SamplerData.h"
+
+// MOOSE includes
+#include "Sampler.h"
+
+template <>
+InputParameters
+validParams<SamplerData>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  params.addClassDescription(
+      "Tool for extracting Sampler object data and storing in VectorPostprocessor vectors.");
+  params += validParams<SamplerInterface>();
+  params.addRequiredParam<SamplerName>("sampler",
+                                       "The sample from which to extract distribution data.");
+  return params;
+}
+
+SamplerData::SamplerData(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters), SamplerInterface(this), _sampler(getSampler("sampler"))
+{
+}
+
+void
+SamplerData::initialize()
+{
+  for (VectorPostprocessorValue * ptr : _sample_vectors)
```

Comment:
Quite simply, people are really bad at guessing types. Nearly every single PR that goes up has at least one incorrectly assigned type. I agree that having types in the source can help with readability because they still describe intention even if they are slightly incorrect.

On the flip side, the more you use `auto` the more comfortable you get with just allowing the interfaces to do their jobs. It also makes refactoring a breeze when everyone uses `auto` as there are few places where you end up changing types. It'll take time, but I think people will get better at naming variables to describe usage (i.e. I really only care if something is numeric, or character based, or a vector. I don't really care if it's an int versus an unsigned long and this is where people make mistakes anyway).